realm/realm-library/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -2278,4 +2278,34 @@ public void run() {
             throw exception.get(0);
         }
     }
+
+    public void testIsEmpty() {
+      RealmConfiguration realmConfig = TestHelper.createConfiguration(getContext(), "empty_test.realm");
+      Realm.deleteRealm(realmConfig);
+      Realm emptyRealm = Realm.getInstance(realmConfig);
+
+      assertTrue(emptyRealm.isEmpty());
+
+      emptyRealm.beginTransaction();
+      PrimaryKeyAsLong obj = new PrimaryKeyAsLong();
+      obj.setId(1);
+      obj.setName("Foo");
+      emptyRealm.copyToRealm(obj);
+      assertFalse(emptyRealm.isEmpty());
+      emptyRealm.cancelTransaction();
+
+      assertTrue(emptyRealm.isEmpty());
+
+      emptyRealm.beginTransaction();
```

Comment:
I think we need this section to check `assertFalse(emptyRealm.isEmpty())` on line 2306.